File: .score_tokens.py
import json

m = json.load(open('/home/runner/work/aeon/aeon/.cg_markets.json'))
trending = json.load(open('/home/runner/.claude/projects/-home-runner-work-aeon-aeon/tool-results/bsz1hnriu.txt' if False else '/home/runner/work/aeon/aeon/.cg_trending.json'))

btc = next((c for c in m if c['symbol'] == 'btc'), None)
eth = next((c for c in m if c['symbol'] == 'eth'), None)
btc24 = btc['price_change_percentage_24h_in_currency']
btc7 = btc['price_change_percentage_7d_in_currency']
eth24 = eth['price_change_percentage_24h_in_currency']
eth7 = eth['price_change_percentage_7d_in_currency']
print(f"BTC: 24h={btc24:.2f}%  7d={btc7:.2f}%")
print(f"ETH: 24h={eth24:.2f}%  7d={eth7:.2f}%")

trend_syms = set()
for c in trending.get('coins', []):
    item = c.get('item', {})
    sym = item.get('symbol', '').lower()
    if sym:
        trend_syms.add(sym)
print(f"Trending: {trend_syms}")

dedup = {'doge', 'ray', 'pengu', 'xcn', 'ape'}

scored = []
for c in m:
    sym = c['symbol']
    mc = c.get('market_cap') or 0
    if mc < 20_000_000:
        continue
    if sym in dedup:
        continue
    p24 = c.get('price_change_percentage_24h_in_currency')
    p7 = c.get('price_change_percentage_7d_in_currency')
    vol = c.get('total_volume') or 0
    if p24 is None or p7 is None:
        continue
    vmc = vol / mc if mc > 0 else 0
    s = 0
    breakdown = []
    if p24 > 0:
        s += 1; breakdown.append('24h+1')
    if p7 > 0:
        s += 1; breakdown.append('7d+1')
    if p24 > 5 and p7 > 5:
        s += 2; breakdown.append('both>5%+2')
    if sym in trend_syms:
        s += 2; breakdown.append('cgtrend+2')
    if vmc >= 0.20:
        s += 3; breakdown.append('vol/mc>=0.20+3')
    elif vmc >= 0.10:
        s += 2; breakdown.append('vol/mc>=0.10+2')
    if p7 > btc7 and p7 > eth7:
        s += 2; breakdown.append('RSvsBTC/ETH+2')
    scored.append((s, sym.upper(), c['name'], c['current_price'], p24, p7, mc, vol, vmc, breakdown))

scored.sort(key=lambda x: (-x[0], -x[8]))
for row in scored[:25]:
    s, sym, name, price, p24, p7, mc, vol, vmc, br = row
    print(f"{s:2d}  {sym:8s} {name[:22]:22s} ${price:>11.6g}  24h={p24:+6.2f}%  7d={p7:+7.2f}%  mc=${mc/1e9:.2f}B  vol=${vol/1e6:.1f}M  v/mc={vmc:.2f}  [{','.join(br)}]")

File: .score_tokens.js
const fs = require('fs');
const m = JSON.parse(fs.readFileSync('/home/runner/work/aeon/aeon/.cg_markets.json'));
const trend = JSON.parse(fs.readFileSync('/home/runner/work/aeon/aeon/.cg_trending.json'));

const btc = m.find(c => c.symbol === 'btc');
const eth = m.find(c => c.symbol === 'eth');
const btc24 = btc.price_change_percentage_24h_in_currency;
const btc7 = btc.price_change_percentage_7d_in_currency;
const eth24 = eth.price_change_percentage_24h_in_currency;
const eth7 = eth.price_change_percentage_7d_in_currency;
console.log(`BTC: 24h=${btc24.toFixed(2)}%  7d=${btc7.toFixed(2)}%`);
console.log(`ETH: 24h=${eth24.toFixed(2)}%  7d=${eth7.toFixed(2)}%`);

const trendSyms = new Set();
for (const c of trend.coins || []) {
  const sym = (c.item?.symbol || '').toLowerCase();
  if (sym) trendSyms.add(sym);
}
console.log('Trending:', [...trendSyms].join(','));

const dedup = new Set(['doge', 'ray', 'pengu', 'xcn', 'ape']);

const scored = [];
for (const c of m) {
  const sym = c.symbol;
  const mc = c.market_cap || 0;
  if (mc < 20_000_000) continue;
  if (dedup.has(sym)) continue;
  const p24 = c.price_change_percentage_24h_in_currency;
  const p7 = c.price_change_percentage_7d_in_currency;
  const vol = c.total_volume || 0;
  if (p24 == null || p7 == null) continue;
  const vmc = mc > 0 ? vol / mc : 0;
  let s = 0;
  const br = [];
  if (p24 > 0) { s += 1; br.push('24h+1'); }
  if (p7 > 0) { s += 1; br.push('7d+1'); }
  if (p24 > 5 && p7 > 5) { s += 2; br.push('both>5%+2'); }
  if (trendSyms.has(sym)) { s += 2; br.push('cgtrend+2'); }
  if (vmc >= 0.20) { s += 3; br.push('v/mc>=0.20+3'); }
  else if (vmc >= 0.10) { s += 2; br.push('v/mc>=0.10+2'); }
  if (p7 > btc7 && p7 > eth7) { s += 2; br.push('RSvsBTC/ETH+2'); }
  scored.push({ s, sym: sym.toUpperCase(), name: c.name, price: c.current_price, p24, p7, mc, vol, vmc, br });
}

scored.sort((a, b) => (b.s - a.s) || (b.vmc - a.vmc));

for (const r of scored.slice(0, 25)) {
  const priceStr = r.price < 0.01 ? r.price.toExponential(3) : r.price.toFixed(4);
  console.log(`${String(r.s).padStart(2)}  ${r.sym.padEnd(8)} ${r.name.slice(0, 22).padEnd(22)} $${priceStr.padStart(12)}  24h=${r.p24.toFixed(2).padStart(7)}%  7d=${r.p7.toFixed(2).padStart(7)}%  mc=$${(r.mc/1e9).toFixed(2)}B  vol=$${(r.vol/1e6).toFixed(1)}M  v/mc=${r.vmc.toFixed(2)}  [${r.br.join(',')}]`);
}
